testing/endtoend/components/lighthouse_validator.go

```diff
@@ -324,46 +325,52 @@ func setupKeystores(valClientIdx, startIdx, numOfKeys int) (string, error) {
 	encryptor := keystorev4.New()
 	// Use lighthouse's default password for their insecure keystores.
 	password := "222222222222222222222222222222222222222222222222222"
-	for i, pk := range pubKeys {
-		pubKeyBytes := pk.Marshal()
-		cryptoFields, err := encryptor.Encrypt(privKeys[i].Marshal(), password)
-		if err != nil {
-			return "", errors.Wrapf(
-				err,
-				"could not encrypt secret key for public key %#x",
-				pubKeyBytes,
-			)
-		}
-		id, err := uuid.NewRandom()
-		if err != nil {
-			return "", err
-		}
-		kStore := &keymanager.Keystore{
-			Crypto:  cryptoFields,
-			ID:      id.String(),
-			Pubkey:  fmt.Sprintf("%x", pubKeyBytes),
-			Version: encryptor.Version(),
-			Name:    encryptor.Name(),
-		}
-
-		fPath := filepath.Join(secretsPath, "0x"+kStore.Pubkey)
-		if err := file.WriteFile(fPath, []byte(password)); err != nil {
-			return "", err
-		}
-		keystorePath := filepath.Join(validatorKeystorePath, "0x"+kStore.Pubkey)
-		if err := file.MkdirAll(keystorePath); err != nil {
-			return "", err
-		}
-		fPath = filepath.Join(keystorePath, "voting-keystore.json")
-		encodedFile, err := json.MarshalIndent(kStore, "", "\t")
-		if err != nil {
-			return "", errors.Wrap(err, "could not marshal keystore to JSON file")
-		}
-		if err := file.WriteFile(fPath, encodedFile); err != nil {
-			return "", err
-		}
+	g, _ := errgroup.WithContext(context.Background())
+	for idx, pkey := range pubKeys {
+		pk := pkey
+		i := idx
```

Comment:
why not `for i, pk := range pubKeys`?

Reply:
oh i see, goroutine enclosure. although you don't need to copy the `idx` value, and you could do `pubKeyBytes := pk.Marshal()` outside the goroutine, which would eliminate the need to copy.